pylint/checkers/base.py

```diff
@@ -2098,6 +2162,52 @@ def _name_became_keyword_in_version(name, rules):
                 return ".".join(str(v) for v in version)
         return None
 
+    @staticmethod
+    def _assigns_typevar(node: Optional[nodes.NodeNG]) -> bool:
+        """Check if a node is assigning a TypeVar"""
+        if isinstance(node, astroid.Call):
+            inferred = utils.safe_infer(node.func)
+            if (
+                inferred
+                and isinstance(inferred, astroid.ClassDef)
+                and inferred.qname() == TYPING_TYPE_VAR_QNAME
+            ):
+                return True
```

Comment:
```suggestion
            return (
                inferred
                and isinstance(inferred, astroid.ClassDef)
                and inferred.qname() == TYPING_TYPE_VAR_QNAME
            )
```

Reply:
For some reason `mypy` fails on this..
Reverting.